src/server/registry/util/util_test.go

```diff
@@ -27,3 +29,185 @@ func TestIndexString(t *testing.T) {
 	assert.True(t, IndexString(a, "Z") == -1)
 	assert.True(t, IndexString(a, "") == -1)
 }
+
+func Test_pickItems(t *testing.T) {
+	p := func(n int) *int {
+		return &n
+	}
+
+	type args struct {
+		items []string
+		n     *int
+		last  string
+	}
+	tests := []struct {
+		name  string
+		args  args
+		want  []string
+		want1 string
+	}{
+		{
+			"no parameters",
+			args{[]string{"a", "b", "c", "d"}, nil, ""},
+			[]string{"a", "b", "c", "d"},
+			"",
+		},
+		{
+			"n=0",
+			args{[]string{"a", "b", "c", "d"}, p(0), ""},
+			[]string{},
+			"",
+		},
+		{
+			"n=1",
+			args{[]string{"a", "b", "c", "d"}, p(1), ""},
+			[]string{"a"},
+			"a",
+		},
+		{
+			"n=2",
+			args{[]string{"a", "b", "c", "d"}, p(2), ""},
+			[]string{"a", "b"},
+			"b",
+		},
+		{
+			"n=4", // n is the count of items
+			args{[]string{"a", "b", "c", "d"}, p(4), ""},
+			[]string{"a", "b", "c", "d"},
+			"",
+		},
+		{
+			"n=5", // n is bigger than the count of items
+			args{[]string{"a", "b", "c", "d"}, p(5), ""},
+			[]string{"a", "b", "c", "d"},
+			"",
+		},
+		{
+			"last=a",
+			args{[]string{"a", "b", "c", "d"}, nil, "a"},
+			[]string{"b", "c", "d"},
+			"",
+		},
+		{
+			"last=d",
+			args{[]string{"a", "b", "c", "d"}, nil, "d"},
+			[]string{},
+			"",
+		},
+		{
+			"n=1 last=a",
+			args{[]string{"a", "b", "c", "d"}, p(1), "a"},
+			[]string{"b"},
+			"b",
+		},
+		{
+			"n=2 last=a",
+			args{[]string{"a", "b", "c", "d"}, p(2), "a"},
+			[]string{"b", "c"},
+			"c",
+		},
+		{
+			"n=3 last=a", // just the left n items
+			args{[]string{"a", "b", "c", "d"}, p(3), "a"},
+			[]string{"b", "c", "d"},
+			"",
+		},
+		{
+			"n=4 last=a", // left items is less than n
+			args{[]string{"a", "b", "c", "d"}, p(4), "a"},
+			[]string{"b", "c", "d"},
+			"",
+		},
+		{
+			"n=1 last=d", // last is the last element of the items
+			args{[]string{"a", "b", "c", "d"}, p(1), "d"},
+			[]string{},
+			"",
+		},
+		{
+			"last=v3", // last element not found
+			args{[]string{"v1", "v2", "v4", "v5"}, nil, "v3"},
+			[]string{"v4", "v5"},
+			"",
+		},
+		{
+			"one item",
+			args{[]string{"a"}, nil, ""},
+			[]string{"a"},
+			"",
+		},
+		{
+			"one item n=2",
+			args{[]string{"a"}, p(2), ""},
+			[]string{"a"},
+			"",
+		},
+		{
+			"two items",
+			args{[]string{"a", "b"}, nil, ""},
+			[]string{"a", "b"},
+			"",
+		},
+		{
+			"three items",
+			args{[]string{"a", "b", "c"}, nil, ""},
+			[]string{"a", "b", "c"},
+			"",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, got1 := pickItems(tt.args.items, tt.args.n, tt.args.last)
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("pickItems() got = %v, want %v", got, tt.want)
+			}
+			if got1 != tt.want1 {
+				t.Errorf("pickItems() got1 = %v, want %v", got1, tt.want1)
+			}
+		})
+	}
+}
+
+func Test_sortedAndUniqueItems(t *testing.T) {
```

Comment:
TestSortedAndUniqueItems

Reply:
This test function name and skeleton are generated by the vscode-go, so let's keep this name.